File: RCSS_Fronted/lib/service_implementation/chat_service.dart
import "dart:convert";

import "package:http/http.dart" as http;

class ChatService {
  Map<String, String> get headers {
    return {
      'Content-Type': 'application/json; charset=UTF-8',
    };
  }

  Future getRecentTalkHistory(
      String uuid, String business_id, String business_service_name) async {
    final uri = Uri.parse(
        "http://10.10.10.207:3000/api/businessServiceTalks/${uuid}/${business_id}/${business_service_name}");
    final response = await http.get(uri);

    Map<String, dynamic> responseBody = json.decode(response.body);

    if (response.statusCode == 200) {
      return responseBody;
    } else {
      throw Exception('${response.statusCode}');
    }
  }
}

File: SignalRChat/SignalRChat/SignalRChat.Client/Service/ChatService.cs
using System.Text.Json;

namespace SignalRChat.Client.Service
{
    public class ChatService
    {
        private readonly HttpClient _httpClient;

        public ChatService(HttpClient httpClient)
        {
            _httpClient = httpClient;
        }

        public class RecentHistoryMessages 
        {
            public int business_id { get; set; }
            public string business_service_name { get; set; }
            public string user_uuid { get; set; }
            public DateTime created_date { get; set; }
            public bool is_user_talk { get; set; }
            public int backend_user_id { get; set; }
            public string talk_content { get; set; }
            public bool is_user_read { get; set; }
            public bool is_backend_user_read { get; set; }
            public string user_name { get; set; }
            public string backend_user_name { get; set; }
        }

        public class RecentHistoryMessagesList 
        {
            public RecentHistoryMessages[] data { get; set; }
            public int code { get; set; }
            public string message { get; set; }
        }

        public async Task<RecentHistoryMessagesList> GetRecentHistoryMessagesList(string uuid, string businessId, string serviceName) 
        {
            try
            {
                var uri = $"http://10.10.10.207:3000/api/businessServiceTalks/{uuid}/{businessId}/{serviceName}";
                var response = await _httpClient.GetAsync(uri);

                if (response.IsSuccessStatusCode)
                {
                    string data = await response.Content.ReadAsStringAsync();
                    RecentHistoryMessagesList jsonData = JsonSerializer.Deserialize<RecentHistoryMessagesList>(data);
                    Console.WriteLine(jsonData.data[0].talk_content);

                    return jsonData;
                }
            }
            catch (Exception ex) 
            {
                Console.WriteLine($"錯誤: {ex.Message}");
            }
            return new RecentHistoryMessagesList();
        }
    }
}
